src/rpc/rawtransaction.cpp

```diff
@@ -578,10 +579,10 @@ static RPCHelpMan decodescript()
 
     std::vector<std::vector<unsigned char>> solutions_data;
     const TxoutType which_type{Solver(script, solutions_data)};
+    const bool is_witness_v0{which_type == TxoutType::WITNESS_V0_KEYHASH || which_type == TxoutType::WITNESS_V0_SCRIPTHASH};
 
-    if (which_type == TxoutType::SCRIPTHASH) {
-        // P2SH cannot be wrapped in a P2SH. If this script is already a P2SH,
-        // don't return the address for a P2SH of the P2SH.
+    if (which_type == TxoutType::SCRIPTHASH || !is_witness_v0) {
```

Comment:
In b9cc2008ecfe823da840dd1c321aa298f8c8bef2 "rpc: Only return P2SH wrapped witness programs for BIP141 types in de… "

This doesn't work as it is also catching all of the legacy things that can be wrapped such as multisig, p2pkh, p2pk, etc.

I think it would be better to just enumerate lists of P2SH wrappable, and P2WSH wrappable, and check against those. I doubt any additional types would be added to those lists in the future.

Reply:
Fixed